src/xs2arr/model.py

```diff
@@ -1,107 +1,397 @@
-        if any(T_grid < 0.0):
-            raise ValueError("All values in T_grid must be >= 0.0")
-
-        # For the Arrhenius equation fittings.
-        params = Parameters()
-        params.add("a", value=1e-14)
-        params.add("b", value=0.1)
-        params.add("c", value=-10.0, max=0.0)
+        T_grid = _validate_and_prepare_T_grid(T_grid, mean_E_grid)
+
+        if not isinstance(logarithmic, bool):
+            raise TypeError("logarithmic must be of type bool")
+
+        regressor, params = _create_fitting_model(logarithmic)
 
         results = []
 
         for cross_section_info in self.cross_section_set.cross_sections:
-            xs_energy = asarray(cross_section_info.data["energy"], dtype=float)
-            xs = asarray(cross_section_info.data["cross section"], dtype=float)
+            xs_energy = np.asarray(cross_section_info.data["energy"], dtype=float)
+            xs = np.asarray(cross_section_info.data["cross section"], dtype=float)
 
             # Interpolate the cross section onto the EEDF grid.
             xs_interp = interpolate_xs(self.eedf_grid, xs_energy, xs)
 
             # Perform the rate integrals.
             rates = [
-                compute_rate(
+                self.rate_computer(
                     self.eedf_grid, xs_interp, self.eedf_cls(T).pdf(self.eedf_grid)
                 )
                 for T in T_grid
             ]
 
-            rates = asarray(rates, dtype=float)
+            rates = np.asarray(rates, dtype=float)
 
             # Convert rates to appropriate units.
-            rates *= sqrt(2.0 * q / m_e)
+            rates *= np.sqrt(2.0 * q / m_e)
 
-            # And compute the a, b and c Arrhenius coefficients.
-            regressor = FittingModel(arrhenius, independent_vars=["T"])
+            T_grid, rates = _remove_bad_data(T_grid, rates, logarithmic)
 
-            results.append(regressor.fit(rates, params, T=T_grid))
+            rates = np.log10(rates) if logarithmic else rates
+
+            fitting = regressor.fit(rates, params, T=T_grid)
+
+            a_true, b_true, c_true = _get_true_abc(fitting, logarithmic)
+
+            results.append((fitting, a_true, b_true, c_true))
 
         return results
+
+
+def _validate_and_prepare_lxcat_file(lxcat_file: str | None):
+    """
+    Validates and prepares the LXCAT data file for cross section analysis.
+
+    Parameters
+    ----------
+    lxcat_file : str | None
+        Path to the LXCAT data file containing cross section information.
+
+    Returns
+    -------
+    CrossSectionSet
+        Parsed cross section data from the LXCAT file.
+
+    Raises
+    ------
+    ValueError
+        If lxcat_file is None or an empty string.
+    TypeError
+        If lxcat_file is not a string.
+    """
+
+    if lxcat_file is None:
+        raise ValueError("No lxcat file provided")
+    if not isinstance(lxcat_file, str):
+        raise TypeError("lxcat_file must be of type str")
```

Comment:
Given that these are the first lines of the function, and that this is the first `validate_and_prepare` in the class `__init__`, why not change the type from `str | None` to just `str` in both? The following check of `if not isinstance(lxcat_file, str)` will catch a `None` anyway, plus we could report any incorrect types to the user:

```suggestion
    if not isinstance(lxcat_file, str):
        raise TypeError(f"lxcat_file must be of type str, found {type(lxcat_file)}")
```